ui-performance.md

```diff
@@ -0,0 +1,470 @@
+displays performance statistics using two graphs.
+The following screenshot shows the performance overlay running on the Flutter
+Gallery example:
+
+<center><img src="images/performance-overlay-green.png" alt="screenshot of performance overlay showing zero jank"></center>
+<center>The graphs produced by the performance overlay.<br>THe vertical green bar represents the current frame.</center>
+
+Each graph represents the last 300 frames for the following two threads used in every Flutter
+app:
+
+* Engine<br>
+  The engine graph (on the bottom) represents time spent executing Dart code in the Dart VM.
+  This thread is sometimes called the UI thread and includes code that you wrote,
+  and code performed by Flutter’s framework on your app’s behalf.
+
+* Rasterizer<br>
+  The rasterizer graph (on the top) represents time spent painting your application,
+  or interfacing with the phone’s GPU (graphics processing unit) thread. When your app creates
+  and displays a scene, the framework creates a “layer tree”, a lightweight object containing
+  device-agnostic painting commands, and sends the layer tree to the GPU thread to be rendered
+  on the device. You cannot directly access the GPU thread or its data but, if this thread is
+  slow, it’s a result of something you’ve done in the Dart code.
+
+Generally, each frame should be created and displayed within 16ms (or 1/60th of a second).
+A frame exceeding this limit fails to display, resulting in jank, and a vertical red bar
+appears in one or both of the graphs. If a red bar appears in the engine graph,
+the Dart code is too expensive. If a red vertical bar appears in the rasterizer graph,
+the scene is too complicated to render quickly.
+
+<center><img src="images/performance-overlay-jank.png" al="Screenshot of performance overlay showing jank."></center>
+<center>The vertical red bar in the rasterier graph indicates that the current frame is expensive to paint.</center>
+
+### Displaying the performance overlay
+
+You can toggle the performance overlay in several ways:
+
+* Using the Flutter Inspector
+* From the command line
+* Programmatically
+
+Flutter Inspector is available in the VSCode, Android Studio, and IntelliJ plugins,
+or at the command line, but its UI may change as it continues to evolve.
+Learn more about what the inspector can do in the
+[Flutter Inspector talk](https://www.youtube.com/watch?v=JIcmJNT9DNI) presented at DartConf 2018.
+
+#### Flutter Inspector
+
+The easiest way to enable the PerformanceOverlay widget is by enabling it in the Flutter
+Inspector, which is available through the Flutter plugin for your IDE. The Inspector
+view opens by default when running an application. If the inspector is not available,
+you can display it as follows.
+
+In Android Studio and IntelliJ IDEA:
+
+1. Select **View > Tool Windows > Flutter Inspector**.
+1. In the gear's menu, select **Show Performance Overlay**.
```

Comment:
This is now accessed from the Inspector toolbar (not the gear menu).

<img width="245" alt="screen shot 2018-03-02 at 7 41 29 pm" src="https://user-images.githubusercontent.com/1269969/36930255-bdddf952-1e51-11e8-9a9b-a11dfd5563d4.png">

The `Toggle Performance Overlay` item.

Reply:
Oh, thanks for letting me know, @devoncarew!